langchain/embeddings/mlflow_gateway.py

```diff
@@ -0,0 +1,50 @@
+from __future__ import annotations
+from functools import lru_cache
+from typing import List, Any, Optional,TYPE_CHECKING
+
+from pydantic import BaseModel
+
+from langchain.embeddings.base import Embeddings
+
+
+if TYPE_CHECKING:
+    import mlflow.gateway
+
+
+def chunk(texts: List[str], size: int) -> List[List[str]]:
+    for i in range(0, len(texts), size):
+        yield texts[i : i + size]
+
+
+@lru_cache()
+def _get_client(gateway_uri: Optional[str]) -> mlflow.gateway.MlflowGatewayClient:
+    import mlflow.gateway
+
+    return mlflow.gateway.MlflowGatewayClient(gateway_uri)
+
+
+class MlflowGatewayEmbeddings(Embeddings, BaseModel):
+    route: str
+    gateway_uri: Optional[str]
+
+    def __init__(self, **kwargs: Any):
+        import mlflow
+
+        super().__init__(**kwargs)
+        mlflow.gateway.set_gateway_uri(self.gateway_uri)
+
+    def _query(self, texts: List[str]) -> List[List[float]]:
+        embeddings = []
+        for t in chunk(texts, 1000):
+            resp = _get_client(self.gateway_uri).query(
+            self.route,
+            data={"text": t},
+        )
```

Comment:
@harupy Can we use a smaller batch size? We're going to need to limit the batch size we support on Databricks to protect the health of the service (avoid OOMs).

@mingyu89 I recall that we discussed a max batch size of 50 initially. Does that still work?

Reply:
https://docs.google.com/document/d/19bTRlKFNDxDczBMebv1aj0fe_eOk5jK4wIrPeX7-WeA/edit
Without streaming support, if we use a buffer size of 640k for each response, we'll be able to support batch size up to 32.
I would recommend 20, 25(or 30) to be safer.

Reply:
Awesome, okay. Let's stick with 20. Best to be safe; we can scale up later.